### ISSUE TITLE
Add support for generic stub/service

### ISSUE THREAD

**Issue by Tradias** [Tradias/asio-grpc#22]
Add convenience overloads to `agrpc::request` and `agrpc::repeatedly_request` for `grpc::GenericServerAsyncReaderWriter`, `grpc::GenericClientAsyncReaderWriter` and the like.

Also consider adding a generic server benchmark to [grpc_bench](https://github.com/Tradias/grpc_bench).

Work is being done on the [generic-rpcs](https://github.com/Tradias/asio-grpc/tree/generic-rpcs) branch.

Targets v1.7.0.

**Comment by Tradias** [comment]
@4eUeP Overloads for `agrpc::request` and `agrpc::repeatedly_request` for generic requests are now available on the above mentioned branch. They should have better performance then the test snippet that I posted in your issue. Let me know if you face any issues.
I am finalizing documentation and then they will be merged into master.

**Comment by 4eUeP** [comment]
@Tradias Great! 

An innocuous problem: building on ubuntu 22.04 provided grpc (v1.30.2) failed because of `error: ‘TemplatedGenericStub’ in namespace ‘grpc’ does not name a template type`. However, v1.44.0 of grpc works. Since the README says, "older versions work as well".

A small suggestion: put the generated docs directory to another branch (e.g. `gh-pages`). The motivation is I want to figure out some functions' definitions by doing a global text search, and the generated docs are always the top results.

BTW, I didn't notice much performance difference between the origin test snippet and the new `agrpc::repeatedly_request`.

**Comment by Tradias** [comment]
Thank you very much for testing it out!

I definitely want to support gRPC v1.30.2 so I will add a task to the CI pipeline and fix the issue regarding `TemplatedGenericStub`.

Good point with the `gh-pages`, for now I have disabled automatic doc updates on feature branches. That should at least make the diff to `master` easier to read and browse. When I was reading about Github Pages, it didn't really mention the option to use a `gh-pages` branch anymore, maybe it is deprecated. I will double check.

Yep, I do not expect much performance difference in `agrpc::repeatedly_request`. It is doing pretty much the same as the test snippet except with a simpler allocation strategy.

**Comment by 4eUeP** [comment]
@Tradias Hi, 

Github Pages says:

> "You can also choose to publish your site from a different branch or folder"
> <https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site>

And, the branch `gh-pages` is usually used and preferred for hosting pages but is not required, any other branches are allowed.

You can find how to configure a different branch by this documentation:
<https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site>

**Comment by Tradias** [comment]
Thanks again, the documentation now lives on the [gh-pages](https://github.com/Tradias/asio-grpc/tree/gh-pages) branch.

Regarding `TemplatedGenericStub`, I figured out that it was first stabilized in v1.31.0 (see [commit](https://github.com/grpc/grpc/commit/f71801ffd6ce4218f32f53bf0f1b2c4e87c0f2e5)). I cannot find a gRPC version macro however, so I replaced the convenience functions for `TemplatedGenericStub` with `GenericStub`. Users can always use `agrpc::grpc_initiate` if they want to use `TemplatedGenericStub` with a type other than `grpc::ByteBuffer`. As a side effect, there is now also a Github Action that tests gRPC v1.16.1, which actually succeeds.

It did however show that the use of `grpc::ProtoBufferWriter writer{&buffer, message_byte_size, message_byte_size}` can lead to an assertion hit in `GPR_CODEGEN_ASSERT(byte_count_ < total_size_)`, I have therefore replaced it with calls to `grpc::GenericSerialize`.
